src/test/url-handling.test.js

```diff
@@ -1366,3 +1367,64 @@ describe('last requested call tree summary strategy', function() {
     );
   });
 });
+
+describe('symbolServerUrl', function() {
+  function setup(search: string) {
+    jest.spyOn(console, 'error').mockImplementation(() => {});
+
+    const { getState } = _getStoreWithURL({ search });
+    const symbolServerUrl = urlStateSelectors.getSymbolServerUrl(getState());
+    const urlState = urlStateSelectors.getUrlState(getState());
+    const queryString = getQueryStringFromUrlState(urlState);
+
+    return {
+      symbolServerUrl,
+      queryString,
+    };
+  }
+
+  it('defaults to the Mozilla symbol server', function() {
+    const { symbolServerUrl, queryString } = setup('');
+    expect(symbolServerUrl).toEqual(SYMBOL_SERVER_URL);
+    expect(symbolServerUrl.substr(-1)).not.toEqual('/');
+    expect(queryString).not.toContain('symbolServer=');
+  });
+
+  it('can be switch to a custom localhost server', function() {
+    const { symbolServerUrl, queryString } = setup(
+      '?symbolServer=http://localhost:1234/symbols'
+    );
+    expect(symbolServerUrl).toEqual('http://localhost:1234/symbols');
+    expect(queryString).toContain(
+      'symbolServer=http%3A%2F%2Flocalhost%3A1234%2Fsymbols'
+    );
+  });
+
+  it('removes any trailing slash', function() {
+    const { symbolServerUrl, queryString } = setup(
+      '?symbolServer=http://localhost:1234/'
+    );
+    expect(symbolServerUrl).toEqual('http://localhost:1234');
+    expect(queryString).toContain(
+      'symbolServer=http%3A%2F%2Flocalhost%3A1234%2F'
+    );
+  });
+
+  it('will error when switching to an unknown host', function() {
+    const { symbolServerUrl, queryString } = setup(
+      '?symbolServer=http://symbols.mozilla.org.example.com/symbols'
+    );
+    expect(symbolServerUrl).toEqual(SYMBOL_SERVER_URL);
+    expect(queryString).toContain(
+      'symbolServer=http%3A%2F%2Fsymbols.mozilla.org.example.com%2Fsymbols'
+    );
+    expect(console.error.mock.calls).toMatchSnapshot();
+  });
+
+  it('will error when switching to an invalid host', function() {
+    const { symbolServerUrl, queryString } = setup('?symbolServer=invalid');
+    expect(symbolServerUrl).toEqual(SYMBOL_SERVER_URL);
+    expect(queryString).toContain('symbolServer=invalid');
+    expect(console.error.mock.calls).toMatchSnapshot();
+  });
```

Comment:
It would be good to add a few tests for the missing cases as outlined by the codecov tool.

If you don't have time to do them I can do it before merging, just tell me :-)

Missing cases:
* valid host but http instead of https (rejected)
* valid host with https (accepted)

Reply:
I'll add those tests.